src/Middleware/HttpLogging/src/HttpLoggingServicesExtensions.cs

```diff
@@ -15,22 +17,37 @@ public static class HttpLoggingServicesExtensions
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
     /// <param name="configureOptions">A delegate to configure the <see cref="HttpLoggingOptions"/>.</param>
-    /// <returns></returns>
+    /// <returns>The original service collection for chaining.</returns>
     public static IServiceCollection AddHttpLogging(this IServiceCollection services, Action<HttpLoggingOptions> configureOptions)
     {
         ArgumentNullException.ThrowIfNull(services);
         ArgumentNullException.ThrowIfNull(configureOptions);
 
+        services.TryAddSingleton(ObjectPool.ObjectPool.Create<HttpLoggingInterceptorContext>());
+        services.TryAddSingleton(TimeProvider.System);
         services.Configure(configureOptions);
         return services;
     }
 
+    /// <summary>
+    /// Registers the given type as a <see cref="IHttpLoggingInterceptor"/> in the DI container.
+    /// </summary>
+    /// <typeparam name="T">A type that implements <see cref="IHttpLoggingInterceptor"/>.</typeparam>
+    /// <param name="services">The service collection.</param>
+    /// <returns>The original service collection for chaining.</returns>
+    public static IServiceCollection AddHttpLoggingInterceptor<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] T>
+        (this IServiceCollection services) where T : class, IHttpLoggingInterceptor
+    {
+        services.TryAddSingleton<IHttpLoggingInterceptor, T>();
```

Comment:
Does this allow adding multiple interceptors? Seems that it's not: https://dotnetfiddle.net/D5yzwP